ipa/verifier.go

```diff
@@ -48,34 +48,27 @@ func CheckIPAProof(transcript *common.Transcript, ic *IPAConfig, commitment band
 		commitment = commit([]banderwagon.Element{commitment, L, R}, []fr.Element{fr.One(), x, xInv})
 	}
 
-	current_basis := ic.SRSPrecompPoints.SRS
-
-	for i := 0; i < len(challenges); i++ {
-
-		G_L, G_R := splitPoints(current_basis)
-
-		b_L, b_R := splitScalars(b)
-
-		xInv := challenges_inv[i]
-
-		b = foldScalars(b_L, b_R, xInv)
-		current_basis = foldPoints(G_L, G_R, xInv)
+	g := ic.SRSPrecompPoints.SRS
+
+	// We compute the folding-scalars for g and b.
+	foldingScalars := make([]fr.Element, len(g))
+	for i := 0; i < len(g); i++ {
+		scalar := fr.One()
+
+		for challengeIdx := 0; challengeIdx < len(challenges); challengeIdx++ {
```

Comment:
We decompose the index `i` ([0,255]) into it's binary form (which matches `len(challenges)`).